File: hospital_management_system/src/main/java/com/synotech/hos/services/PatientService.java
package com.synotech.hos.services;

import java.util.List;

import com.synotech.hos.model.Location;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.synotech.hos.dao.PatientRepo;
import com.synotech.hos.model.Patient;

@Service
public class PatientService {
	
	private final PatientRepo patientRepo;
	
	@Autowired
	public PatientService(PatientRepo patientRepo) {
		this.patientRepo = patientRepo;
	}
	
	public List<Patient> getPatients(){
		return patientRepo.findAll();
	}

	public List<Patient> getPatientForTable(){

		return patientRepo.getPatientForTable();
	}
	public Patient save(Patient newPatient) {

		return patientRepo.save(newPatient);
	}

	public void transfer(Integer location, Integer id){
		Integer admit = 2;
		 patientRepo.updateLocation(location, id, admit);
	}

	public void admit(Integer location, Integer id, Integer wid){
		 int admit = 1;
		patientRepo.updateWard(location, id, wid, admit);
	}

	public void discharge(Integer id){
		int admit = 0;
		patientRepo.discharge(id, admit);
	}

	public List<Patient> getPatientsForAdmit(){
		return patientRepo.getPatientForAdmit();
	}

	public List<Patient> getPatientsForDischrge(){
		return patientRepo.getPatientsForDischarge();
	}
}

File: hospital_management_system_frt/src/service/PatientService.js
import axios from "axios";
// import authHeader from "./auth.headers";

class PatientService {
    getPatients() {
        return axios.get('http://localhost:8081/api/patient')
    }

    getPatientsForAdmit() {
        return axios.get('http://localhost:8081/api/patient/admit')
    }

    getPatientsForDischarge() {
        return axios.get('http://localhost:8081/api/patient/discharge')
    }

    updatePatientLocation(location, id, wid) {
        return axios.put('http://localhost:8081/api/patient/admit/' + parseInt(location) + '/' + parseInt(id) + '/' + parseInt(wid), {
            location: parseInt(location),
            id: parseInt(id)
        })
    }

    dischargePatient(id) {
        return axios.put('http://localhost:8081/api/patient/discharge/' + parseInt(id), {

        })
    }

    transferPatient(location, id) {
        return axios.put('http://localhost:8081/api/patient/' + parseInt(location) + '/' + parseInt(id), {

        })
    }
}

export default new PatientService()